lib/sudo-exec/src/monitor.rs

```diff
@@ -58,8 +56,25 @@ impl MonitorRelay {
             let command_pgrp = command_pid;
             setpgid(command_pid, command_pgrp);
 
+            macro_rules! set_signal_handler {
```

Comment:
with a `Signal` enum this could be a for loop instead of a macro

Reply:
I used a macro because the callbacks have to be function pointers instead of closures to avoid borrowing issues in the implementation of the event loop. Even if `Signal` were an `enum` we'd need to capture the signal value, wouldn't we?